dotnet/src/SemanticKernel.Abstractions/Orchestration/ContextVariables.cs

```diff
@@ -59,30 +56,7 @@ public ContextVariables Clone()
     /// <returns>The current instance</returns>
     public ContextVariables Update(string? value)
     {
-        this._variables[MainKey] = value ?? string.Empty;
-        return this;
-    }
-
-    /// <summary>
-    /// Updates all the local data with new data, merging the two datasets.
-    /// Do not discard old data
-    /// </summary>
-    /// <param name="newData">New data to be merged</param>
-    /// <param name="merge">Whether to merge and keep old data, or replace. False == discard old data.</param>
-    /// <returns>The current instance</returns>
-    public ContextVariables Update(ContextVariables newData, bool merge = true)
-    {
-        if (!object.ReferenceEquals(this, newData))
-        {
-            // If requested, discard old data and keep only the new one.
-            if (!merge) { this._variables.Clear(); }
-
-            foreach (KeyValuePair<string, string> varData in newData._variables)
-            {
-                this._variables[varData.Key] = varData.Value;
-            }
-        }
-
+        this.Set(MainKey, value);
```

Comment:
Allowing entry for `MainKey` to be set to null here but not in the constructor. Behaviour should be consistent.

Reply:
@markwallace-microsoft. According with the `Set` implementation, setting `null` will remove the variable.